### DISCUSSION TITLE
Where are the individual parameters of different models stored?

### DISCUSSION THREAD

**Discussion by at-src** [juanmc2005/diart#244] [Q&A] [answered]
I am looking for the hyperparameter values of pyannote/segmentation model (onset, offset, min_duration_on, min_duration_off,etc) but not able to find them. How are they related to diart parameters tau, rho, and delta? I am getting different timestamp values for the same audio by using pyannote/segmentation as vad in isolation and using diart. I want both to have same timestamp values.
Example for the same audio,
output of pyannote/segmentation is 
Speech detected: 1.30s - 7.05s (Duration: 5.75s)
Speech detected: 7.32s - 15.44s (Duration: 8.12s)
Speech detected: 15.86s - 17.14s (Duration: 1.28s)
Speech detected: 17.63s - 32.60s (Duration: 14.97s)
Speech detected: 32.80s - 37.24s (Duration: 4.44s)
Speech detected: 37.32s - 40.33s (Duration: 3.00s)
Speech detected: 41.43s - 49.10s (Duration: 7.68s)
Speech detected: 49.36s - 54.77s (Duration: 5.42s)
Speech detected: 56.48s - 58.17s (Duration: 1.69s)
and diart is
speaker0 1.33-16.51
speaker1 16.51-17.06
speaker0 17.65-18.51
speaker2 18.51-21.51
speaker0 21.51-22.51
speaker3 22.51-32.51
speaker2 33.01-38.01
speaker3 38.01-41.51
speaker0 41.51-58.18

**Comment by juanmc2005** [comment, marked as answer]
Hi @krane5555, diart uses pyannote.audio models for streaming, where as their pipelines are for offline pre-recorded audio. Even though the models might be the same, the algorithms putting everything together are very different, so the results will most certainly not be the same. Streaming performance may approach offline performance as the latency is increased though.

For an explanation of hyper-parameters, see #133

**Reply by at-src** [reply]
So is there any way to have same timestamps from a vad (possibly integrated in a stt model) and diart both in real time? I think segmentation model is responsible for timestamp generation in diart and timestamp generation mechanism in stt is dependent on vad so i think it will be difficult to have both on same page or making it same across both for seamless integration.

**Reply by juanmc2005** [reply]
If you absolutely want to keep VAD timestamps, you can always use diart's output to determine which speakers are present in each VAD segment
